src/contrib/coh_gamma/COHNCGamma_ext/NCgamma_Form_Factors.cpp

```diff
@@ -0,0 +1,212 @@
+        nu=double(i);
+        aux += pow(-1.0,nu+1.0)*R*R*R*a[i]*hc*hc*hc/(pi*pi*nu*nu-Q*Q*R*R);
+    }
+
+    this->FF_p = 4.0*pi*sin(Q*R)/(Q*R)*aux;// /6.0;
+    this->FF_n = this->FF_p;
+
+}
+
+void Nucleus_FF_DeVries::set_12C() {
+    //a(i) en fm^-3
+    this->a[1]=0.15737e-1;
+    this->a[2]=0.38897e-1;
+    this->a[3]=0.37085e-1;
+    this->a[4]=0.14795e-1;
+    this->a[5]=-0.44834e-2;
+
+    this->a[6]=-0.10057e-1;
+    this->a[7]=-0.68696e-2;
+    this->a[8]=-0.28813e-2;
+    this->a[9]=-0.77229e-3;
+    this->a[10]=0.66908e-4;
+
+    this->a[11]=0.10636e-3;
+    this->a[12]=-0.36864e-4;
+    this->a[13]=-0.50135e-5;
+    this->a[14]=0.9455e-5;
+    this->a[15]=-0.47687e-5;
+
+    this->R=8.0/hc; // [fm]/hc
+}
+
+void Nucleus_FF_DeVries::set_40Ar() {
+    //a(i) en fm^-3
+    this->a[1]=0.30451e-1;
+    this->a[2]=0.55337e-1;
+    this->a[3]=0.20203e-1;
+    this->a[4]=-0.16765e-1;
+    this->a[5]=-0.13578e-1;
+
+    this->a[6]=-0.43204e-4;
+    this->a[7]=0.91988e-3;
+    this->a[8]=-0.41205e-3;
+    this->a[9]=0.11971e-3;
+    this->a[10]=-0.19801e-4;
+
+    this->a[11]=-0.43204e-5;
+    this->a[12]=0.61205e-5;
+    this->a[13]=-0.37803e-5;
+    this->a[14]=0.18001e-5;
+    this->a[15]=-0.77407e-6;
+
+    this->R=9.0/hc; // [fm]/hc
+}
+
+Nucleus_FF_DeVries::~Nucleus_FF_DeVries() {}
+
+
+
+/////////////////////////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////////////////////////
+
+
+
+
+Form_Factors_Delta::Form_Factors_Delta() {
+    this->mn = NCgamma_Param::mp;
+    this->mn2 = NCgamma_Param::mp2;
+    this->mDelta = NCgamma_Param::mDelta;
+    this->mDelta2 = this->mDelta*this->mDelta;
+    this->N_Delta_MA = NCgamma_Param::N_Delta_MA;
+    this->aNC = 1.0 - 2.0*NCgamma_Param::sinw2;
+
+    this->kgcm0 = (this->mDelta2-this->mn2)/2.0/this->mDelta;
+    this->mpw2 = (this->mn+this->mDelta)*(this->mn+this->mDelta);
+    this->mmw2 = (this->mn - this->mDelta)*(this->mn - this->mDelta);
+
+    this->C4v = 0.0;
+    this->C5v = 0.0;
+    this->C3a = 0.0;
+    this->C4a = 0.0;
+    this->C5a = 0.0;
+    this->C6a = 0.0;
+
+    this->C4vNC = 0.0;
+    this->C5vNC = 0.0;
+    this->C3aNC = 0.0;
+    this->C4aNC = 0.0;
+    this->C6aNC = 0.0;
+
+}
+
+void Form_Factors_Delta::setFF(double Q2) {
+
+    double egcm = (this->mDelta2-Q2-this->mn2)/2.0/this->mDelta;
+    double qcm = sqrt(egcm*egcm + Q2);
+
+    double Fq = 1.0/((1.0 + Q2/NCgamma_Param::parameter_071)*(1.0 + Q2/NCgamma_Param::parameter_071));
+    double AM = NCgamma_Param::parameter_03*(1.0 + NCgamma_Param::parameter_001*Q2)*exp(-NCgamma_Param::parameter_023*Q2)*(qcm/this->kgcm0)*Fq;
+    double A32= sqrt(3.0)/2.0*(-AM);
+
+    double r = sqrt(2.0*this->kgcm0/NCgamma_Param::pi/NCgamma_Param::alpha*this->mn*this->mDelta/(this->mmw2+Q2));
+
+    this->C3v=-r*this->mn*this->mDelta/(this->mpw2+Q2)*(2.0*A32);
+
+    double Fd=pow((1.0 + Q2/(this->N_Delta_MA*this->N_Delta_MA)),-2);
+    this->C5aNC=1.2*Fd;
```

Comment:
All the next comments are related to the code given to us by Eduardo. 
I totally agree with all your comments and apart from the last `1.2`  we managed to make those parameters consistent with the rest of genie or having them configurable. 

Since this is the actual code we received I think we should keep it like that.